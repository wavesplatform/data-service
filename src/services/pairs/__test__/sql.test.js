const sql = require('../sql');

describe('sql query from pairs', () => {
  it('should get one pair', () => {
    expect(
      sql.get({
        amountAsset: '111',
        priceAsset: '222',
      })
    ).toMatchSnapshot();
  });

  it('should get many pairs', () => {
    expect(
      sql.mget([
        {
          amountAsset: '111',
          priceAsset: '222',
        },
        {
          amountAsset: '333',
          priceAsset: '444',
        },
      ])
    ).toMatchSnapshot();
  });

  it('should search pairs', () => {
    expect(sql.search({ limit: 1 })).toMatchSnapshot();
  });
});
