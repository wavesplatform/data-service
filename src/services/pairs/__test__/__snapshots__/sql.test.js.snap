// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql query from pairs should get many pairs 1`] = `"select * from \\"pairs\\" as \\"t\\" where (\\"t\\".\\"amount_asset_id\\", \\"t\\".\\"price_asset_id\\") in (('111', '222'), ('333', '444'))"`;

exports[`sql query from pairs should get one pair 1`] = `"select * from \\"pairs\\" as \\"t\\" where (\\"t\\".\\"amount_asset_id\\", \\"t\\".\\"price_asset_id\\") in (('111', '222'))"`;

exports[`sql query from pairs should search pairs 1`] = `"select * from \\"pairs\\" as \\"t\\" order by t.volume_waves desc NULLS LAST limit 1"`;
